### PR TITLE
add lookup param for request

### PR DESCRIPTION
**Pull request by machao**
http://nodejs.cn/api/http.html#http_http_request_url_options_callback

usage:

```
needle({
  url,
  lookup: (hostname, options, callback) => {
     if(hostname.endsWith('.localhost')) {
        setImmediate(() => {
           callback(null, '127.0.0.1', 4);
        });
        return
     }
     dns.lookup(hostname, options, callback);
  }
})
```

### CODE REVIEW

File: lib/needle.js [hunk 1]
```diff
@@ -16,7 +16,8 @@ var fs          = require('fs'),
     auth        = require('./auth'),
     cookies     = require('./cookies'),
     parsers     = require('./parsers'),
-    decoder     = require('./decoder');
+    decoder     = require('./decoder'),
+    dns         = require('dns');
```
Comment (tomas) [inline lib/needle.js:20]: Is this necessary?

**Comment by lyswhut** [comment]
I need this option and hope to merge soon

File: lib/needle.js (hunk 1 above)
Comment (lyswhut) [reply lib/needle.js:20]: It doesn't seem to be necessary

Comment (tomas) [reply lib/needle.js:20]: @machao could you update the PR?

**Comment by lyswhut** [comment]
@machao Has this option been added?
If not, can you help add this option?
Thank!

**Comment by machao** [comment]
i'm sorry about useless code in this pr, and i open new one. #360

**Pull request closed without merging**